dds/idl/value_reader_generator.cpp

```diff
@@ -150,6 +157,12 @@ namespace {
       return;
     }
 
+    // TODO Not sure how to check for annotations here.
+    if (actual->annotation_appls().find("@optional") != nullptr) {
```

Comment:
Technically optional has a boolean parameter that defaults to true, so even though it doesn't make much sense to me, someone could say `@optional(FALSE)` and this would treat that the same as `@optional` or `@optional(TRUE)`.

Also in `opendds_idl` to simplify things like that we define and interact with annotations indirectly. The annotations are defined in [`annotations.h`](https://github.com/OpenDDS/OpenDDS/blob/master/dds/idl/annotations.h) and then have functions in [`be_global`](https://github.com/OpenDDS/OpenDDS/blob/f4ec88a04e0cf45bbf4257bad6abb3ddf0b5a771/dds/idl/be_global.h#L227) for getting values. Since we already have `@optional` itself setup, here you'd want:

```suggestion
    if (be_global->is_optional(actual)) {
```

Reply:
I tried using that initially but it never entered the `if`, although I wasn't using `actual` I was using the `type` if that makes a difference

Reply:
I realized this function is for a whole type. `@optional`, as far as I know, can only apply to struct members. The check and probably the inserting of `begin_optional` and `end_optional` should go around here:

https://github.com/OpenDDS/OpenDDS/blob/f4ec88a04e0cf45bbf4257bad6abb3ddf0b5a771/dds/idl/value_reader_generator.cpp#L314

Passing `field` to `is_optional` should work.